### ISSUE TITLE
_leftAxisMax/Min, YMax/Min, _rightAxisMax/Min of data should not be changed based on VisibleXRange

### ISSUE THREAD

**Issue by aelam** [danielgindi/Charts#2187]
When `autoScaleMinMaxEnabled` is true 
autoScale() could change there values based on the visible x range.  

I'm thinking `ChartData` doesn't need to know what the visible x-range is. and the XMin, XMax, YMin, YMax of data shouldn't be effected by visible range of ChartView

the Left/RightMin/Max  should be stored in `ChartView` or a brand-new class which is a property of `ChartView`
`ChartData`  can still provide an API for calculating the minMaxY_leftAxisMax/Min YMax/Min, _rightAxisMax/Min based on visibleXRange but don't store them in it

**Comment by liuxuan30** [comment]
Hey @aelam  https://github.com/danielgindi/Charts/pull/2177 fixed your issue right?

**Comment by aelam** [comment]
No. It could be a designing improvement. Just got this idea from the bug #2177
